Source/WebGPU/WebGPU/Buffer.mm

```diff
@@ -479,6 +482,21 @@ static size_t computeRangeSize(uint64_t size, size_t offset)
     indirectBufferRecomputed(0, 0, 0, 0, MTLIndexTypeUInt16, 0);
 }
 
+#if HAVE(WEBGPU_SWIFT)
+void Buffer::copy(const std::span<const uint8_t> data, const size_t offset)
+{
+    auto buffer = getBufferContents();
+    RELEASE_ASSERT(buffer);
+    auto endOffset = checkedSum<size_t>(offset, data.size());
+    RELEASE_ASSERT(!(endOffset.hasOverflowed() || endOffset.value() > currentSize()));
+    auto checkSize = checkedSum<size_t>(currentSize());
+    RELEASE_ASSERT(!checkSize.hasOverflowed());
+    auto destination = std::span<uint8_t> { buffer + offset, static_cast<size_t>(currentSize()) - offset };
+    WTFLogAlways("NITIN %s:%d", __FILE__, __LINE__);
+    WebGPU::copySpan(destination, data);
+}
+#endif
```

Comment:
It's fine for this patch to be minimal -- but the obvious next step is to move this code into the Swift WebGPU module, so that you get all the null checking and overflow checking automatically.